user-service/src/main/java/by/battle/userservice/service/UserServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package by.battle.userservice.service;
+
+import by.battle.userservice.dto.UserDto;
+import by.battle.userservice.entity.Status;
+import by.battle.userservice.entity.User;
+import by.battle.userservice.exception.UserNotFoundException;
+import by.battle.userservice.mapper.UserDtoMapper;
+import by.battle.userservice.repository.RoleRepository;
+import by.battle.userservice.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
+    private final UserDtoMapper userDtoMapper;
+
+    @Override
+    public List<UserDto> findAll() {
```

Comment:
Service should not work with Dto's, this should be a businell layer, which is working with core logic and base entities.

Controller - application layer
Service - business layer
Repository - Data access layer

so controller should be responsible to map from/to DTO's
Service - should perform application logic.
Repository - work with database.

read more:

https://en.wikipedia.org/wiki/Multitier_architecture